### PR TITLE
chore(inputs.net_response): Use Init to check configuration issues

### PR DESCRIPTION
**Pull request by Hipska**
Plugin cleanup so users don't receive configuration errors every gather cycle:
image

### CODE REVIEW

**Comment by Hipska** [comment]
There are 2 tests which aren't working, could somebody help me out how to fix them? I think it has to do with the fixed port number?

**Comment by powersj** [comment]
@Hipska - you should turn on the ability to let maintainers push to your repo :) because that isn't turned on here is a diff you can use:

```diff
diff --git a/plugins/inputs/net_response/net_response.go b/plugins/inputs/net_response/net_response.go
index 034ee953a..90809d4cd 100644
--- a/plugins/inputs/net_response/net_response.go
+++ b/plugins/inputs/net_response/net_response.go
@@ -206,7 +206,6 @@ func (n *NetResponse) Init() error {
 // It will call either UDPGather or TCPGather based on the configuration and
 // also fill an Accumulator that is supplied.
 func (n *NetResponse) Gather(acc telegraf.Accumulator) error {
-
        // Prepare host and port
        host, port, err := net.SplitHostPort(n.Address)
        if err != nil {
diff --git a/plugins/inputs/net_response/net_response_test.go b/plugins/inputs/net_response/net_response_test.go
index 0b7f01800..fa3b8201b 100644
--- a/plugins/inputs/net_response/net_response_test.go
+++ b/plugins/inputs/net_response/net_response_test.go
@@ -83,7 +83,7 @@ func TestTCPError(t *testing.T) {
                        "result_type": "connection_failed",
                },
                map[string]string{
-                       "server":   "",
+                       "server":   "localhost",
                        "port":     "9999",
                        "protocol": "tcp",
                        "result":   "connection_failed",
@@ -211,7 +211,7 @@ func TestUDPError(t *testing.T) {
                },
                map[string]string{
                        "result":   "read_failed",
-                       "server":   "",
+                       "server":   "localhost",
                        "port":     "9999",
                        "protocol": "udp",
                },
```

**Comment by Hipska** [comment]
I believe it was enabled when creating the PR, but don't see this option anymore after creation.

Anyway, the suggested changes do not fix the problems.

**Comment by Hipska** [comment]
Nevermind, it seems like I did something wrong.

**Review by powersj — APPROVED** [review]

File: plugins/inputs/net_response/net_response.go
Comment (srebhan) [inline plugins/inputs/net_response/net_response.go:213]: I think this could also be done in `Init()`, but I don't want to hold up the PR just for this...

File: plugins/inputs/net_response/net_response_test.go
Comment (srebhan) [inline plugins/inputs/net_response/net_response_test.go:24]: You might want to use `require.ErrorContains` here.

**Review by srebhan — APPROVED** [review]
Thanks for the improvement @Hipska! I left two small comments for educational purposes mostly :-) but I'm merging the PR now. Feel free to apply the suggestions in another PR.

File: plugins/inputs/net_response/net_response_test.go
Comment (Hipska) [reply plugins/inputs/net_response/net_response_test.go:24]: Oh didn't know, thanks.

File: plugins/inputs/net_response/net_response.go
Comment (Hipska) [reply plugins/inputs/net_response/net_response.go:213]: It is done in Init to be able to error out. But should the `host` and `port` then be stored in the struct maybe?

**Pull request merged**
